### DISCUSSION TITLE
The refresh rate of the notebook

### DISCUSSION THREAD

**Discussion by Sen0E** [seerge/g-helper#786] [Q&A]
When I have a second monitor, the refresh rate of the laptop screen will change to the refresh rate of my second screen, not the refresh rate of the laptop screen, is this normal
This is the refresh rate for the second monitor
![image](https://github.com/seerge/g-helper/assets/98098882/c741580b-bcfe-4ba5-8074-e3711e249c20)
This is the refresh rate of the laptop
![image](https://github.com/seerge/g-helper/assets/98098882/90ca0175-de1e-4034-8da4-55c90a33d848)

**Comment by seerge** [comment]
@zSXHY can you post a log from ``%appdata%/ghelper`` here ?

**Reply by Sen0E** [reply]
2023/7/15 20:50:12: WMI event 192
2023/7/15 20:50:19: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:50:20: Screen = 240Hz : -1
2023/7/15 20:50:21: Screen = 240Hz : -1
2023/7/15 20:50:21: ScreenOverdrive = 0 : OK
2023/7/15 20:50:21: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:50:21: Screen = 240Hz : -1
2023/7/15 20:50:22: Screen = 240Hz : -1
2023/7/15 20:50:22: ScreenOverdrive = 0 : OK
2023/7/15 20:50:22: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:50:22: Screen = 240Hz : -1
2023/7/15 20:50:24: Screen = 240Hz : -1
2023/7/15 20:50:24: ScreenOverdrive = 0 : OK
2023/7/15 20:50:24: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:50:24: Screen = 240Hz : -1
2023/7/15 20:50:25: Screen = 240Hz : -1
2023/7/15 20:50:25: ScreenOverdrive = 0 : OK
2023/7/15 20:50:27: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:50:28: Screen = 240Hz : -1
2023/7/15 20:50:28: Screen = 240Hz : -1
2023/7/15 20:50:28: ScreenOverdrive = 0 : OK
2023/7/15 20:50:50: WMI event 192
2023/7/15 20:51:07: 
==============NVSMI LOG==============

Timestamp                                 : Sat Jul 15 20:51:06 2023
Driver Version                            : 528.49
CUDA Version                              : 12.0

Attached GPUs                             : 1
GPU 00000000:01:00.0
    Product Name                          : NVIDIA GeForce RTX 4070 Laptop GPU
    Product Brand                         : GeForce
    Product Architecture                  : Ada Lovelace
    Display Mode                          : Enabled
    Display Active                        : Disabled
    Persistence Mode                      : N/A
    MIG Mode
        Current                           : N/A
        Pending                           : N/A
    Accounting Mode                       : Disabled
    Accounting Mode Buffer Size           : 4000
    Driver Model
        Current                           : WDDM
        Pending                           : WDDM
    Serial Number                         : N/A
    GPU UUID                              : GPU-a5a5ea99-5075-1523-8809-bb98fd82b622
    Minor Number                          : N/A
    VBIOS Version                         : 95.06.15.40.33
    MultiGPU Board                        : No
    Board ID                              : 0x100
    Board Part Number                     : N/A
    GPU Part Number                       : 2860-775-A1
    Module ID                             : 1
    Inforom Version
        Image Version                     : G002.0000.00.03
        OEM Object                        : 2.0
        ECC Object                        : N/A
        Power Management Object           : N/A
    GPU Operation Mode
        Current                           : N/A
        Pending                           : N/A
    GSP Firmware Version                  : N/A
    GPU Virtualization Mode
        Virtualization Mode               : None
        Host VGPU Mode                    : N/A
    IBMNPU
        Relaxed Ordering Mode             : N/A
    PCI
        Bus                               : 0x01
        Device                            : 0x00
        Domain                            : 0x0000
        Device Id                         : 0x286010DE
        Bus Id                            : 00000000:01:00.0
        Sub System Id                     : 0x25DD1043
        GPU Link Info
            PCIe Generation
                Max                       : 4
                Current                   : 4
                Device Current            : 4
                Device Max                : 4
                Host Max                  : 4
            Link Width
                Max                       : 8x
                Current                   : 8x
        Bridge Chip
            Type                          : N/A
            Firmware                      : N/A
        Replays Since Reset               : 0
        Replay Number Rollovers           : 0
        Tx Throughput                     : 2000 KB/s
        Rx Throughput                     : 0 KB/s
        Atomic Caps Inbound               : N/A
        Atomic Caps Outbound              : N/A
    Fan Speed                             : N/A
    Performance State                     : P0
    Clocks Throttle Reasons
        Idle                              : Not Active
        Applications Clocks Setting       : Not Active
        SW Power Cap                      : Not Active
        HW Slowdown                       : Not Active
            HW Thermal Slowdown           : Not Active
            HW Power Brake Slowdown       : Not Active
        Sync Boost                        : Not Active
        SW Thermal Slowdown               : Not Active
        Display Clock Setting             : Not Active
    FB Memory Usage
        Total                             : 8188 MiB
        Reserved                          : 226 MiB
        Used                              : 0 MiB
        Free                              : 7962 MiB
    BAR1 Memory Usage
        Total                             : 8192 MiB
        Used                              : 1 MiB
        Free                              : 8191 MiB
    Compute Mode                          : Default
    Utilization
        Gpu                               : 0 %
        Memory                            : 0 %
        Encoder                           : 0 %
        Decoder                           : 0 %
    Encoder Stats
        Active Sessions                   : 0
        Average FPS                       : 0
        Average Latency                   : 0
    FBC Stats
        Active Sessions                   : 0
        Average FPS                       : 0
        Average Latency                   : 0
    Ecc Mode
        Current                           : N/A
        Pending                           : N/A
    ECC Errors
        Volatile
            SRAM Correctable              : N/A
            SRAM Uncorrectable            : N/A
            DRAM Correctable              : N/A
            DRAM Uncorrectable            : N/A
        Aggregate
            SRAM Correctable              : N/A
            SRAM Uncorrectable            : N/A
            DRAM Correctable              : N/A
            DRAM Uncorrectable            : N/A
    Retired Pages
        Single Bit ECC                    : N/A
        Double Bit ECC                    : N/A
        Pending Page Blacklist            : N/A
    Remapped Rows
        Correctable Error                 : 0
        Uncorrectable Error               : 0
        Pending                           : No
        Remapping Failure Occurred        : No
        Bank Remap Availability Histogram
            Max                           : 64 bank(s)
            High                          : 0 bank(s)
            Partial                       : 0 bank(s)
            Low                           : 0 bank(s)
            None                          : 0 bank(s)
    Temperature
        GPU Current Temp                  : 49 C
        GPU T.Limit Temp                  : 26 C
        GPU Shutdown Temp                 : 100 C
        GPU Slowdown Temp                 : 97 C
        GPU Max Operating Temp            : 75 C
        GPU Target Temperature            : N/A
        Memory Current Temp               : N/A
        Memory Max Operating Temp         : N/A
    Power Readings
        Power Management                  : Supported
        Power Draw                        : 17.48 W
        Power Limit                       : 4294967.50 W
        Default Power Limit               : 100.00 W
        Enforced Power Limit              : 105.00 W
        Min Power Limit                   : 5.00 W
        Max Power Limit                   : 140.00 W
    Clocks
        Graphics                          : 2085 MHz
        SM                                : 2085 MHz
        Memory                            : 7780 MHz
        Video                             : 2025 MHz
    Applications Clocks
        Graphics                          : N/A
        Memory                            : N/A
    Default Applications Clocks
        Graphics                          : N/A
        Memory                            : N/A
    Deferred Clocks
        Memory                            : N/A
    Max Clocks
        Graphics                          : 3105 MHz
        SM                                : 3105 MHz
        Memory                            : 8001 MHz
        Video                             : 2415 MHz
    Max Customer Boost Clocks
        Graphics                          : N/A
    Clock Policy
        Auto Boost                        : N/A
        Auto Boost Default                : N/A
    Voltage
        Graphics                          : 925.000 mV
    Fabric
        State                             : N/A
        Status                            : N/A
    Processes                             : None


2023/7/15 20:51:07: Stopped: RadeonSoftware
2023/7/15 20:51:07: Running eco command 1
2023/7/15 20:51:07: GPUEco = 1 : OK
2023/7/15 20:51:07: Eco flag : 1
2023/7/15 20:51:07: Mux flag : 1
2023/7/15 20:51:07: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:51:07: Screen = 240Hz : OK
2023/7/15 20:51:07: ScreenOverdrive = 0 : OK
2023/7/15 20:51:11: Running eco command 0
2023/7/15 20:51:12: GPUEco = 0 : OK
2023/7/15 20:51:12: Eco flag : 0
2023/7/15 20:51:12: Mux flag : 1
2023/7/15 20:51:12: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY1
2023/7/15 20:51:12: Screen = 240Hz : OK
2023/7/15 20:51:12: ScreenOverdrive = 0 : OK
2023/7/15 20:51:17: NVIDIA GeForce RTX 4070 Laptop GPU
2023/7/15 20:51:17: GET GPU CLOCKS: 0, 0
2023/7/15 20:51:17: SET GPU CLOCKS: -200, -220
2023/7/15 20:51:34: Varibright: 0,0
2023/7/15 20:51:46: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY22
2023/7/15 20:51:46: LI2041 Wide:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY1
2023/7/15 20:51:46: Screen = 60Hz : OK
2023/7/15 20:51:46: ScreenOverdrive = 0 : OK
2023/7/15 20:51:48: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \\.\DISPLAY22
2023/7/15 20:51:48: LI2041 Wide:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \\.\DISPLAY1
2023/7/15 20:51:48: Screen = 75Hz : OK
2023/7/15 20:51:48: ScreenOverdrive = 0 : OK
2023/7/15 20:52:06: Varibright: 0,0

**Reply by seerge** [reply]
It looks like that windows (by some reason) is not returning correct refresh rate for a display (by some unknown reason). It could be nvidia optimus interfering, try to set Dispay mode in nvidia settings to Optimus.

**Reply by Sen0E** [reply]
![image](https://github.com/seerge/g-helper/assets/98098882/29e2ecc6-5569-4e55-96b3-34b331969f25)
Here's how I set it up

**Reply by seerge** [reply]
Try to reboot, does it still stay like that?

**Reply by Sen0E** [reply]
It is still the same result, I wonder if it is nvidia driver cause, because I have not encountered this problem before, I found in the Windows display setup is correct

**Reply by seerge** [reply]
@zSXHY well, problem that windows seem to mix displays

```
2023/7/15 20:51:48: NE173QHM-NZ2:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_INTERNAL: \.\DISPLAY22
2023/7/15 20:51:48: LI2041 Wide:DISPLAYCONFIG_OUTPUT_TECHNOLOGY_HDMI: \.\DISPLAY1
```

``\.\DISPLAY1`` - is your internal screen i think, and it swaps it with external monitor ``\.\DISPLAY22`` :) 

Try this build (but most probably result will be same)
[GHelper.zip](https://github.com/seerge/g-helper/files/12062088/GHelper.zip)

And also try to set nvidia display mode to nvidia GPU only and back to optimus after (just to reset it mb)
